### ISSUE TITLE
[AI] [Can't repro] AI won't repair a specific leak at all

### ISSUE THREAD

**Issue by Krzeszny** [Regalis11/Barotrauma#6458]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
The video shows the debugai overlay
[https://youtu.be/28GJaneLhRw](url)

**Steps To Reproduce**
As seen in the video. I couldn't reproduce it.

**Version**
v0.14.6.0 (recorded 10 VII 2021)

**Comment by itchyOwl** [comment]
That's not vanilla Humpback (compare the docking port area and the waypoints): ![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/df3622c0-f73c-480b-96eb-40c1e093ecc5)

Nevertheless fixing leaks should work on that sub too. The bots won't fix leaks _autonomously_ unless the leaks are reported. The bots report them when they see them, but in the first video, you don't report the leaks and there are nobody around to see them.

In the second video the bots indeed ignore the leak. Can't say for sure what's causing it, but when they ignore a leak, they won't target it unless reordered. It's not optimal, but I haven't yet figured out how to solve it. They need to ignore leaks that they can't reach and since those leaks can remain there for long, we can't periodically clear the ignored list. We could periodically recheck the leaks, but checking them without actually going there is not simple, unless it's just a pathing issue. It's worth considering nevertheless. In that video too you have altered the submarine. I wonder if the flipped railgun has something to do with the issue. Did you try to reissue the order? Reissuing clears the ignored targets.

I'm not sure which leak you referred in the third video, but sometimes it happens that the bots can't get to the leak. For example, they might not have access to a suit. Or it might be a waypoint issue. Or there's a pathing issue that could be due to a bug in the code. Nevertheless, they need to ignore those targets or they are stuck. As said above, currently the solution for refreshing the ignored targets is to reissue the order. A better solution would be nice to have in the future.

**Comment by Krzeszny** [comment]
1. Right, I forgot I modified the Humpback for the first video to see what AI does.
2. I can't provide you more info. As I said, it's not always reproducible and I also don't know what's causing it.
3. 
> I'm not sure which leak you referred in the third video, but sometimes it happens that the bots can't get to the leak. For example, they might not have access to a suit. Or it might be a waypoint issue. Or there's a pathing issue that could be due to a bug in the code. Nevertheless, they need to ignore those targets or they are stuck. As said above, currently the solution for refreshing the ignored targets is to reissue the order. A better solution would be nice to have in the future.

In the third video the leak was on the left side of the docking port and they're trying to repair it from below. I think what's happening is AI deciding to repair it from below instead of from that ladder and maybe the welding tool's range isn't enough.

**Comment by itchyOwl** [comment]
Yeah I think they try to target that leak from below, because the closest waypoint is there. Welding tool range might not be enough for that. It's an edge case that I think should be solved by adding that waypoint above the platform.
